src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -231,15 +226,13 @@ public void print(String msg) {
                         },
                         new FileSystem(),
                         new Environment());
-        try (JFRConnection connection = tk.connect(tk.createServiceURL("localhost", 0))) {
-            String id = connection.getJvmId();
-            log.info("Computed self JVM ID: {}", id);
-            return id;
-        } catch (IOException
-                | ReflectionException
-                | MBeanException
-                | InstanceNotFoundException
-                | AttributeNotFoundException e) {
+        try {
+            try (JFRConnection connection = tk.connect(tk.createServiceURL("localhost", 0))) {
+                String id = connection.getJvmId();
+                log.info("Computed self JVM ID: {}", id);
+                return id;
+            }
+        } catch (Exception e) {
```

Comment:
Is this nested `try` intentional?

Reply:
Unfortunately yes, the implicit `connection.close()` on the try-with-resources can actually throw on its own too, so the outer try is to catch from that.

Reply:
Ah, I see. Thanks for the info!